### PR TITLE
Add support for users.getForUser GitHub API

### PR DESCRIPTION
**Pull request by owlishDeveloper**
I need this for badges. Badges make use of users.getForUser GitHub API. I think the best thing would be to add "username" key to user objects in this.github_users of FakeGitHub, and to re-use the existing setUser method.

### CODE REVIEW

File: test/github-auth.js [hunk 1]
```diff
@@ -68,6 +68,15 @@ class FakeGithub {
       'integrations.getInstallationRepositories': async() => {
         return this.repositories;
       },
+      'users.getForUser': async ({username}) => {
+        let user = _.find(this.github_users, {username});
+        if (user) {
+          user.id = parseInt(user.id, 10);
```
Comment (djmitche) [inline test/github-auth.js:74]: Why `parseInt` here after the explicit `toString` call below?

**Review by djmitche — APPROVED** [review]
I like nice, small PRs :)

File: test/github-auth.js (hunk 1 above)
Comment (owlishDeveloper) [reply test/github-auth.js:74]: Well, in real github payloads id is always a number. users.getForId uses id in form of a string - I have no idea why :) That's, I suspect, where toString comes from? ;) But for users.getForUser I need a number.

Comment (owlishDeveloper) [reply test/github-auth.js:74]: I think I should add a comment about that in the code

Comment (djmitche) [reply test/github-auth.js:74]: It looks like `user.getForId` returns the value as an integer (https://api.github.com/users/djmitche).  And `setUser` is just a test function, so if you always pass it an integer, there's no need to do anything.

Comment (djmitche) [reply test/github-auth.js:74]: Oh, I see, users.getById (which already existed) searches by string.  OK, that makes sense then.

Comment (owlishDeveloper) [reply test/github-auth.js:74]: Yeah, I meant get**By**Id, sorry for confusion :)

**Pull request merged**